### DISCUSSION TITLE
Getting Started for an Inexperienced Owner

### DISCUSSION THREAD

**Discussion by astewartnc** [tagyoureit/nodejs-poolController#814] [Q&A]
I have had a Pi that I have used for gaming, so I have some familiarity. I also have a Pentair EasyTouch 4. Our system is very simple: heater, Intelliflo pump, Intellibrite lights. I do have the pump set on three circuits so I have 3 different speeds throughout the day. 

I have read the wiki, I have picked out a good 485 adapter from the recommendations (thank you). But I have questions before I go all-in:
1. Using nodejs-poolController and Dashboard, would I be able to control the system via my phone? Do people just save a chrome bookmark?
2. Is there a particular RPi that folks recommend? 
3. Are there any photos or diagrams to make the connections happen? I have searched and I haven't seen anything. That's what make me the most nervous. 
4. Following up on #3, are there specific wires I need to buy to connect the 485 to my main EasyTouch 4 board?
5. Assuming after I have nJs all up and running, I can still use my main EasyTouch panel, at the same time, correct?
Thank you all! I don't want to spend the $600 for a Pentair controller when I mainly just want to use this for the heater and the lights (my pump is easy enough to set and forget from the main panel)

**Comment by CodeSlinger69** [comment]
Welcome to our hobby :)

> 1. Using nodejs-poolController and Dashboard, would I be able to control the system via my phone? Do people just save a chrome bookmark?

Anything with a browser will work. I just leave a tab open on my device.

> 2. Is there a particular RPi that folks recommend?

I had a 3B laying around in my box of stuff, so I used that. My journey is [here](https://github.com/tagyoureit/nodejs-poolController/discussions/801).

But you don't need a RPi to make this work if all you are doing is controlling your OCP with a better interface. I used njsPC in that mode for about 8 months, and I was running the njsPC & dashPanel software on a Linux VM on my desktop computer. It will run on native Linux and Windows machines too of course.

If you choose to use a RPi, you don't need a lot of processing power. My 3B runs at 1-3% utilization, and that's with a Nixie build.

> 3. Are there any photos or diagrams to make the connections happen? I have searched and I haven't seen anything. That's what make me the most nervous.

Here's a picture of my Easytouch 4P with the RS485 bus connection points circled in green. Find something similar on your board, and connect 4 wires from the RS485 connector to your chosen RS485 adapter. Make sure you get the +15V and the ground correct above all. Those pins are marked on the board if you look carefully, and also in your adapter's documentation, and they are usually connected with red and black wires respectively. The data lines (A and B) are wishy-washy in my experience (and others') - try them one way, if it doesn't work, swap them around.

![et4p](https://github.com/tagyoureit/nodejs-poolController/assets/10177469/efc30df2-d933-42f6-b10c-ff6815a69042)

> 4. Following up on [config.json missing #3](https://github.com/tagyoureit/nodejs-poolController/issues/3), are there specific wires I need to buy to connect the 485 to my main EasyTouch 4 board?

I don't think that issue has anything to do with wiring in an RS485 adapter. You only need 4 wires going from the board to the adapter - any 18-22AWG is fine. I used some extra speaker wire that I had laying around, as you can see in the picture above.

> 5. Assuming after I have nJs all up and running, I can still use my main EasyTouch panel, at the same time, correct?

Definitely.

**Comment by astewartnc** [comment]
> > 2. Is there a particular RPi that folks recommend?
> 
> I had a 3B laying around in my box of stuff, so I used that. My journey is [here](https://github.com/tagyoureit/nodejs-poolController/discussions/801).
> 
> But you don't need a RPi to make this work if all you are doing is controlling your OCP with a better interface. I used njsPC in that mode for about 8 months, and I was running the njsPC & dashPanel software on a Linux VM on my desktop computer. It will run on native Linux and Windows machines too of course.
> 
> If you choose to use a RPi, you don't need a lot of processing power. My 3B runs at 1-3% utilization, and that's with a Nixie build.
> 

First, thank you so much. the picture, more than anything, is a massive help. I do not have any wifi or connectivity out at the ET4 panel right now, so that's why I figured a RPi was absolutely necessary (my router is about 95 feet away), correct? Given my main goal here is to wirelessly control my lights and heater, of course. 

One more Q: Have you or anyone else had success on getting njs working with Google Home? I have read mixed things so not sure what the answer is. Man if I could simply say "Hey Google turn on the pool lights" that would be amazing.

**Reply by Grauwulf75** [reply]
Depending on your pad setup, you can always buy a poweline ethernet adapter set and use that to get your network out to the pad. They aren't necessarily a fast network connection, but you really don't need something super fast to support NjsPC setup. Probably going to be more stable and reliable than a wifi connection at that distance.

**Reply by astewartnc** [reply]
I thought about that - but running a line would be brutal given where my router has to be. I have a solid mesh system that gives me very strong wifi out by the pool (I video call for work out there just fine). 

The RPi + 485 would give me the wifi I need to light controlling right (with njs pool control + dashboard)?

**Reply by CodeSlinger69** [reply]
If you have good Wi-Fi at the pad, then you could use a RS485 to Wi-Fi bridge. That’s what I did, I picked the Elfin EW11. I had to run the antenna out of the metal Easytouch box of course, because metal blocks Wi-Fi. The powerline Ethernet is a great idea too. 

If you can get network connectivity to your pad in some way, then run njsPC on a PC in your house. Simplifies some of the complexity of doing it on an RPi, and it’s more performant, and you don’t need to worry about heat, etc.

**Reply by astewartnc** [reply]
That is true - but unless the PC is on, I wouldnt have controls to the pool, right? That would hurt my intent. I really just want to get wireless controls for my lights, more than anything. 

I will have to keep digging into it. Sadly Pentair's Intelliconnect upgrade from EasyTouch is over $1000. That's a lot, just to get wifi (and ScreenLogic is trash and old I hear)

**Reply by CodeSlinger69** [reply]
Pentair's stuff is very overpriced for what you get. And what @rstrouse and @tagyoureit built here is so much better, so much more flexible, and so much more extensible than anything Pentair can do. 

To your question, njsPC and dashPanel will let you control your lights from your phone (or whatever). dashPanel is basically a web server that serves up those pages to your phone. But you need to be running that web server on a computer somewhere. Whether you do it on a RPi, or on a VM like I did, is up to you, and the particulars of your situation. But yes, that computer will have to be on, whether that computer is a RPi, a Windows PC, a Linux PC, or a VM.

**Comment by Grauwulf75** [comment]
> The RPi + 485 would give me the wifi I need to light controlling right (with njs pool control + dashboard)?

Honestly, I don't know enough about the system yet to say absolutely one way or the other, but it probably will. I'm just new to this and learning/planning out my install, and discovering that my Jandy equipment won't work with NjsPC over RS485 without also having one of the Jandy automation systems as well, which kind of defeats the purpose for me. But I had thought about the network connection prior to diving in, and the powerline adapters were my backup plan. Thankfully my mesh wifi reaches my equipment shed with no issues.
